tests/unit/acquisition/test_rule.py

```diff
@@ -186,3 +189,37 @@ def test_trust_region_for_unsuccessful_local_to_global_trust_region_reduced() ->
     npt.assert_array_less(current_state.eps, previous_state.eps)  # current TR smaller than previous
     assert current_state.is_global
     npt.assert_array_almost_equal(current_state.acquisition_space.lower, lower_bound)
+
+
+class _BatchModelMinusMeanMaximumSingleBuilder(BatchAcquisitionFunctionBuilder):
+    def prepare_acquisition_function(
+        self, dataset: Mapping[str, Dataset], model: Mapping[str, ProbabilisticModel]
+    ) -> BatchAcquisitionFunction:
+        return lambda at: -tf.reduce_max(model[OBJECTIVE].predict(at)[0], axis=-2)
+
+
+def test_batch_acquisition_returns_batches_of_right_size() -> None:
+    search_space = Box(tf.constant([-2.2, -1.0]), tf.constant([1.3, 3.3]))
+    num_query_points = 3
+    ego = BatchAcquisitionRule(
+        num_query_points,
+        _BatchModelMinusMeanMaximumSingleBuilder())
+    dataset = Dataset(tf.zeros([0, 2]), tf.zeros([0, 1]))
+    query_point, _ = ego.acquire(
+        search_space, {OBJECTIVE: dataset}, {OBJECTIVE: QuadraticWithUnitVariance()}
+    )
+    assert query_point.shape == [num_query_points, 2]
```

Comment:
this test is redundant now you've changed the `assert_allclose` in the other

Reply:
this looks more like a unit test than the other one. Why not keep both? or add `assert query_point.shape == [num_query_points, 2]` to the second one? or would `assert_allclose` let us know whereas it's a value problem or a shape problem?

Reply:
`assert_allclose` does shape checks too
```
>>> np.testing.assert_allclose(tf.constant([[1, 1], [1, 1]]), [[1, 1]])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/joel.berkeley/.venv/trieste/joelberkeley/dev/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 1533, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/home/joel.berkeley/.venv/trieste/joelberkeley/dev/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 765, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

(shapes (2, 2), (1, 2) mismatch)
 x: array([[1, 1],
       [1, 1]], dtype=int32)
 y: array([[1, 1]])
```